Dockerfile

```diff
@@ -0,0 +1,41 @@
+# Build stage
+FROM mcr.microsoft.com/dotnet/sdk:7.0.400-bookworm-slim AS build
+WORKDIR /build
+
+COPY *.sln ./
+COPY **/*.csproj ./
+
+RUN dotnet sln list | grep ".csproj" \
+    | while read -r line; do \
+    mkdir -p $(dirname $line); \
+    mv $(basename $line) $(dirname $line); \
+    done;
+
+RUN dotnet restore --use-current-runtime
+
+COPY . .
+
+RUN dotnet publish Refresh.GameServer -c Release --property:OutputPath=/build/publish/ --no-restore --use-current-runtime --self-contained
+
+# Final running container
+
+FROM mcr.microsoft.com/dotnet/runtime:7.0.10-bookworm-slim AS final
+
+# Add non-root user
+RUN set -eux && \
+apt update && \
+apt install -y gosu && \
+rm -rf /var/lib/apt/lists/* && \
+gosu nobody true && \
```

Comment:
Most of this I can follow, but I don't understand this invocation of gosu

Reply:
It’s checking that the gosu binary works by running /bin/true as the nobody user. Gosu is used in many docker entrypoint scripts rather than sudo because sudo can cause weird TTY and signal forwarding behavior